common/djangoapps/student/tests/test_tasks.py

```diff
@@ -51,3 +55,87 @@ def test_send_email(self, mock_log):
             exc_info=True
         )
         self.assertEquals(mock_log.error.call_count, 1)
+
+
+class MockSailthruResponse(object):
+    """
+    Mock object for SailthruResponse
+    """
+
+    def __init__(self, json_response, error=None, code=1):
+        self.json = json_response
+        self.error = error
+        self.code = code
+
+    def is_ok(self):
+        """
+        Return true of no error
+        """
+        return self.error is None
+
+    def get_error(self):
+        """
+        Get error description
+        """
+        return MockSailthruError(self.error, self.code)
+
+
+class MockSailthruError(object):
+    """
+    Mock object for Sailthru Error
+    """
+
+    def __init__(self, error, code=1):
+        self.error = error
+        self.code = code
+
+    def get_message(self):
+        """
+        Get error description
+        """
+        return self.error
+
+    def get_error_code(self):
+        """
+        Get error code
+        """
+        return self.code
+
+
+class SailthruTests(TestCase):
```

Comment:
Can you add a test case if the enrollment mode is `verified`, we **do not** call Sailthru api purchase API?